testsuite/test_framework/git.py

```diff
@@ -15,3 +15,41 @@ def run(self, command) -> RunResult:
     def last(self, limit: int = 1) -> Generator[str, None, None]:
         for i in range(limit):
             yield self.run(["rev-parse", f"HEAD~{i}"]).unwrap().decode().strip()
+
+    def branch(self) -> str:
+        return self.run(["rev-parse", "--abbrev-ref", "HEAD"]).unwrap().decode().strip()
+
+    def branch_exists(self, branch: str) -> bool:
+        return self.run(["rev-parse", "--verify", branch]).succeeded()
+
+    def is_clean(self) -> bool:
+        """Check if the git working directory is clean, using git status --porcelain"""
+        ret = self.run(["status", "--porcelain"])
+        succ = ret.succeeded()
+        out = ret.unwrap().decode().strip()
+        out_empty = len(out) == 0
+        return succ and out_empty
+
+    def remote_matches(self, remote: str, ref: str) -> bool:
+        """Check if the current branch matches the remote branch"""
+        # git ls-remote --heads  https://github.com/aptos-labs/aptos-core.git rustielin/exp
+        remote_commit_hash = self.get_remote_commit_hash(remote, ref)
+        local_commit_hash = self.get_commit_hash(ref)
+        return remote_commit_hash == local_commit_hash
+
+    def get_remote_commit_hash(self, remote: str, ref: str) -> str:
+        try:
+            return (
+                self.run(["ls-remote", "--heads", remote, ref])
+                .unwrap()
+                .decode()
+                .strip()
+                .split()[0]
+            )
+        except IndexError as e:
```

Comment:
I'm not a huge fan of exceptions for non exceptional behavior, i prefer LBYL in this case and or a return status like optional that can handle failure gracefully but thats my style nit